### PR TITLE
Feat: delete placements

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
TIS21-1325

### CODE REVIEW

File: src/main/java/uk/nhs/hee/trainee/details/service/PlacementService.java [hunk 1]
```diff
@@ -70,4 +72,30 @@
...
+
+    if (traineeProfile == null) {
+      return false;
+    }
+    List<Placement> placements = traineeProfile.getPlacements();
+
+    hasDeleted = placements.removeIf(p -> p.getTisId().equals(placementTisId));
+
+    if (hasDeleted) {
+      traineeProfile.setPlacements(placements);
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/details/service/PlacementService.java:93]: Not a major issue, but it should modify the profile's existing list so setting it again shouldn't be needed.

File: src/main/java/uk/nhs/hee/trainee/details/api/PlacementResource.java [hunk 2]
```diff
@@ -68,4 +70,30 @@ public PlacementResource(PlacementService service, PlacementMapper mapper) {
...
+        placementTisId, traineeTisId);
+    try {
+      boolean foundTraineeAndPlacement =
+          service.deletePlacementForTrainee(traineeTisId, placementTisId);
+      if (!foundTraineeAndPlacement) {
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Trainee or placement not found.");
+      }
+    } catch (IllegalArgumentException | InvalidDataAccessApiUsageException e) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getLocalizedMessage());
+      // other exceptions are possible, e.g. DataAccessException if MongoDB is down
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/details/api/PlacementResource.java:95]: Could you add a TODO here to go back and add global exception handlers for these types of exceptions?

File: src/main/java/uk/nhs/hee/trainee/details/service/PlacementService.java (hunk 1 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/trainee/details/service/PlacementService.java]: Of course, thanks for catching this :)

File: src/main/java/uk/nhs/hee/trainee/details/api/PlacementResource.java (hunk 2 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/trainee/details/api/PlacementResource.java:95]: Done

**Review by Judge40 — APPROVED** [review]

**Pull request merged**